pybigquery/sqlalchemy_bigquery.py

```diff
@@ -160,9 +233,28 @@ def has_table(self, connection, table_name, schema=None):
         except NoSuchTableError:
             return False
 
+    def _get_columns_helper(self, columns, cur_columns):
+        """
+        Recurse into record type and return all the nested field names.
+        As contributed by @sumedhsakdeo on issue #17
+        """
+        results = []
+        for col in columns:
+            if col.field_type == 'RECORD':
```

Comment:
Please add `and col.mode != 'REPEATED'` to this line. It's not possible to select nested columns in an array of structures without extra calls to `UNNEST`

Reply:
@tswast sure. good point.

Thinking more, should we change this code to ignore the name of the record field. And return only leaf field names in the record field.
```
+           if col.field_type == 'RECORD' and col.mode != 'REPEATED':
+                cur_columns.append(col)
+                results += self._get_columns_helper(col.fields, cur_columns)
+                cur_columns.pop()
+            else: <<<<<<<<
+                 results += [SchemaField(
+                            name='.'.join(col.name for col in cur_columns + [col]),
+                            field_type=col.field_type,
+                            mode=col.mode,
+                            description=col.description,
+                            fields=col.fields)]
```

Reply:
> return only leaf field names in the record field.

Makes sense to me. It'd probably be weird to have both the record fields and the subfields.

Reply:
@mxmzdlv could you please incorporate this diff in the _get_columns_helper(). Thanks.